### DISCUSSION TITLE
Questions

### DISCUSSION THREAD

**Discussion by bytelinker** [ckjellson/textalloc#7] [General]
Hello,
thank you providing this package. If got some questions:

Could you provide some information in the documentation why to provide

" x_scatter: (array-like), default None
    x-coordinates of all scattered points.
y_scatter: (array-like), default None
    y-coordinates of all scattered points.
"

and how this values are used, please?

Are the x- and y-lims of the axes adjusted or has it to be done by the user? 

Did you try to commit it to matplotlib so that it is included in the package?

Regards

byte-linker

**Comment by ckjellson** [comment]
Hi,

I added short explanation in the Readme. Essentially x_scatter and y_scatter defines positions of points that the text-boxes should not overlap with, and do not have to be the same as the positions of the text-boxes themselves. The algorithm then makes sure that these points are not inside the text-box rectangles (within some margin).

The lims do not change when running allocate_text() by design, and if you want to modify these I suggest you do it before running allocate_text(). Note that you should plot scattered points or lines etc. before calling the function as in the examples.

I have not tried to contribute to matplotlib, and I am not sure if that would be useful and/or reasonable.

Hope that helps!
